src/misc.jl

```diff
@@ -201,3 +201,55 @@ end
     root --> :top
     GraphPlot((source, destiny))
 end
+
+
+@userplot AndrewsPlot
+
+"""
+    andrewsplot(data::AbstractArray; kind=[])
+
+https://en.wikipedia.org/wiki/Andrews_plot
+
+Example
+```julia
+using RDatasets, PlotRecipes
+iris   = dataset("datasets", "iris")
+y      = Array(iris[:,1:4])
+kind   = Array(iris[:,5])
+andrewsplot(y, kind=kind)
+```
+"""
+andrewsplot
+
+@recipe function f(h::AndrewsPlot; kind=[])
+    y = h.args[1]
+    delete!(d, :group)
+    seriestype --> :path
+    title --> "Andrew's plot"
+    label := ""
+    rows,cols = size(y)
+    serieslength = 200
+    ys     = zeros(serieslength,rows)
+    t      = linspace(-π,π,serieslength)
+    sinmat = [sin((i÷2).*ti) for i = 2:cols, ti=t]
+    for j in 1:rows, ti = eachindex(t)
+        ys[ti,j] = y[j,1]/sqrt(2) + sum(y[j,i].*sinmat[i-1,ti] for i = 2:cols)
+    end
+    if isempty(kind)
+        @series begin
+            t,ys
+        end
+    else
+        ugroup = unique(kind)
+        for (gi,g) in enumerate(ugroup)
+            @series begin
+                groupinds = g .== kind
```

Comment:
A clever solution with a bit of massaging the code. This may not be the clearest (it is normally better to pass some of the functionality to a series recipe) but after playing a little with it this morning that isn't easy.